error: ‘strncat’ specified bound 1 equals source length [-Werror=stringop-overflow=]

GCC 11.1.0, Arch Linux x86_64.

```
[ 74%] Building C object src/ponyc/CMakeFiles/ponyc.dir/main.c.o
/build/ponyc/src/ponyc-0.43.0/src/ponyc/main.c: In function ‘main’:
/build/ponyc/src/ponyc-0.43.0/src/ponyc/main.c:94:5: error: ‘strncat’ specified bound 1 equals source length [-Werror=stringop-overflow=]
   94 |     strncat((char*)opt.all_args, " ", 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/ponyc/main.c
+++ b/src/ponyc/main.c
@@ -91,7 +91,17 @@ int main(int argc, char* argv[])
   for (int i = 1; i < argc; i++)
   {
     strncat((char*)opt.all_args, argv[i], size_left);
+    #ifdef __GNUC__
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wstringop-overflow"
+    #endif
+
     strncat((char*)opt.all_args, " ", 1);
+
+    #ifdef __GNUC__
+    #pragma GCC diagnostic pop
+    #endif
+
     size_left -= strlen(argv[i]) + 1;
   }
 

